src/tox_travis/hooks.py

```diff
@@ -11,13 +11,12 @@ def tox_addoption(parser):
         '--travis-after', dest='travis_after', action='store_true',
         help='Exit successfully after all Travis jobs complete successfully.')
 
-    default_toxenv()
-
 
 @tox.hookimpl
 def tox_configure(config):
     """Check for the presence of the added options."""
     if config.option.travis_after:
-        travis_after_monkeypatch()
+        travis_after_monkeypatch(config)
 
     override_ignore_outcome(config)
+    default_toxenv(config)
```

Comment:
> We need to check to make sure that we have tests covering uses of toxenv -e, as well as any default TOXENV, to make sure that we don't override those. Those are escape hatches when all else fails.

`-e` usage was my one concern. There are not currently any tests for it. The code does exit if `TOXENV` is set, but there is no similar check for -e at the moment. 

> If we end up that we're going to use the config everywhere, then I think we should just go ahead and make override_ignore_outcome be called from default_toxenv. It was only ever split because it needed to happen in a different hook.

Makes sense to me. 

> Extra thought: should the hooks just be implemented in the toxenv and after modules? Pluggy lets there be lots of hook implementations, so it seems like it would be reasonable, and would keep them more separated.

Hm. Yes and no. My current thought is that they should remain together since they are essentially the main entry point into the plugin. From here, you can just follow the code. It's not quite as clear when they exist in separate modules.